### DISCUSSION TITLE
Having issues connecting machine learning on a different machine

### DISCUSSION THREAD

**Discussion by Eirikr70** [immich-app/immich#4048] [Q&A]
Hello, I have tries the trick to set up machine learning on a different machine, but it doesn't seem to work : after a few minutes, and although I have given a name to a face, the machine learning container had still 0 activity and memory use. Maybe I should precise that I have set PUID and PGID to my personal user on both machines ?...
Here is my `.env`. I have copied it on both machines.
```
# You can find documentation for all the supported env variables at https://immich.app/docs/install/environment-variables

# The location where your uploaded files are stored
UPLOAD_LOCATION=/srv/nas/immich

# The Immich version to use. You can pin this to a specific version like "v1.71.0"
IMMICH_VERSION=release

# Connection secrets for postgres and typesense. You should change these to random passwords
TYPESENSE_API_KEY=xxx
DB_PASSWORD=the_same_xxx

# The values below this line do not need to be changed
###################################################################################
DB_HOSTNAME=immich_postgres
DB_USERNAME=postgres
DB_DATABASE_NAME=immich

REDIS_HOSTNAME=immich_redis

PUID=1000
PGID=1000

IMMICH_MACHINE_LEARNING_URL=http://192.168.1.104:3003
#IMMICH_MACHINE_LEARNING_ENABLED=true
```
Or should I set the `IMMICH_MACHINE_LEARNING_URL `variable somewhere in the `docker-compose.yml `on the main machine ?

**Comment by bo0tzz** [comment]
> although I have given a name to a face, the machine learning container had still 0 activity and memory use.

What are you expecting to happen? Just setting a name for a face doesn't trigger any machine learning action.

**Reply by Eirikr70** [reply]
Well, since the machine learning container is quite a memory hog, I expected that it would take a bit more than just 25 KB ...
